src/server/project.ts

```diff
@@ -1780,6 +1788,46 @@ namespace ts.server {
         getIncompleteCompletionsCache() {
             return this.projectService.getIncompleteCompletionsCache();
         }
+
+        /*@internal*/
+        getNoDtsResolutionProject(rootFileNames: readonly string[]): Project {
+            if (!this.noDtsResolutionProject) {
+                const options: CompilerOptions = {
+                    ...this.getCompilerOptions(),
+                    noDtsResolution: true,
+                    allowJs: true,
+                    maxNodeModuleJsDepth: 3,
+                    diagnostics: false,
+                    skipLibCheck: true,
+                    sourceMap: false,
+                    types: ts.emptyArray,
+                    lib: ts.emptyArray,
+                    noLib: true,
+                };
+                this.noDtsResolutionProject = new AuxiliaryProject(this.projectService, this.documentRegistry, options);
+            }
+
+            enumerateInsertsAndDeletes<NormalizedPath, NormalizedPath>(
+                rootFileNames.map(toNormalizedPath),
+                this.noDtsResolutionProject.getRootFiles(),
+                compareStringsCaseInsensitive,
+                pathToAdd => {
+                    const info = this.noDtsResolutionProject!.getScriptInfo(pathToAdd);
+                    if (info) {
+                        this.noDtsResolutionProject!.addRoot(info, pathToAdd);
+                    }
+                },
+                pathToRemove => {
+                    // It may be preferable to remove roots only once project grows to a certain size?
+                    const info = this.noDtsResolutionProject!.getScriptInfo(pathToRemove);
+                    if (info) {
+                        this.noDtsResolutionProject!.removeRoot(info);

```

Comment:
can you add a test where file opened from this project moves to inferred project and out of it correctly..

Reply:
Hm, I had forgotten about this part of making the AutoImportProviderProject, but for that one, a ScriptInfo can be part an InferredProject _and_ the AutoImportProviderProject since the latter is something used in the background; it shouldn’t change the experience for a user opening a random file that happens to be part of the auto-imports processing. I think we want to do the same thing for this kind of project.

Reply:
If i remember correctly it might need some extra work as it checks if scriptinfo is part of any project all all.. (containingProjects) on script info so if we havent done something special it might result in some wierd state
The state would be interesting is when you open file thats in AutoImport or auxillary project 

There is also some place where we try to refresh(collect script info or something when we close some other file ) and see if file needs to be moved out or in to inferred project and that would need update too.. because all of those used containingProjects.length or something like that.

Reply:
I believe I handled all of this for AutoImportProvider; it’s pretty simple to update it.

Reply:
Added a unit test that covers at least some of this if you want to take a look.